src/Parser.zig

```diff
@@ -4257,11 +4265,15 @@ fn mulExpr(p: *Parser) Error!Result {
                     else => unreachable,
                 };
             } else {
-                lhs.val = switch (lhs.val) {
-                    .unsigned => |v| .{ .unsigned = v % rhs.val.unsigned },
-                    .signed => |v| .{ .signed = @rem(v, rhs.val.signed) },
-                    else => unreachable,
-                };
+                if (lhs.val == .signed and lhs.val.signed == std.math.minInt(i64) and rhs.val.signed == -1) {
+                    lhs.val = .unavailable;
```

Comment:
Setting the value to `.unavailable` needs to be special cased for macros but here setting it to zero seems to match clang/GCC behavior in:
```c
#if (-9223372036854775807LL - 1LL) % -1 != 0
#error foo
#endif
```